cognite/neat/graph/transformers/_value_type.py

```diff
@@ -0,0 +1,110 @@
+from typing import cast
+
+from rdflib import XSD, Graph, URIRef
+
+from cognite.neat.issues._base import IssueList
+from cognite.neat.rules.analysis import InformationAnalysis
+from cognite.neat.rules.models._rdfpath import SingleProperty
+from cognite.neat.rules.models.data_types import AnyURI, DataType
+from cognite.neat.rules.models.entities import (
+    ClassEntity,
+    MultiValueTypeInfo,
+)
+from cognite.neat.rules.models.information import InformationRules
+
+from ._base import BaseTransformer
+
+
+class SplitMultiValueProperty(BaseTransformer):
+    description: str = (
+        "SplitMultiValueProperty is a transformer that splits a "
+        "multi-value property into multiple single-value properties."
+    )
+    _use_only_once: bool = True
+    _need_changes = frozenset({})
+
+    _object_property_template: str = """SELECT ?s ?o WHERE{{
+
+                                ?s a <{subject_uri}> .
+                                ?s <{property_uri}> ?o .
+                                ?o a <{object_uri}> .
+
+                            }}"""
+
+    _datatype_property_template: str = """SELECT ?s ?o WHERE {{
+
+                                ?s a <{subject_uri}> .
+                                ?s <{property_uri}> ?o .
+                                FILTER (datatype(?o) = <{datatype}>)
+
+                                }}"""
+
+    def transform(self, graph: Graph) -> None:
+        rules = self._infer_rules(graph)
+
+        class_reference_pairs = {class_.class_: class_.reference for class_ in rules.classes}
+
+        for property_ in InformationAnalysis(rules).multi_value_properties:
+            transformation = property_.transformation
+            if transformation and isinstance(transformation.traversal, SingleProperty):
+                traversal = transformation.traversal
+
+                for value_type in cast(MultiValueTypeInfo, property_.value_type).types:
+                    property_uri = rules.prefixes.get(traversal.property.prefix, rules.metadata.namespace)[
+                        traversal.property.suffix
+                    ]
+
+                    subject_uri = rules.prefixes.get(traversal.class_.prefix, rules.metadata.namespace)[
+                        traversal.class_.suffix
+                    ]
```

Comment:
These can be moved outside the loop.
```suggestion
                property_uri = rules.prefixes.get(traversal.property.prefix, rules.metadata.namespace)[
                    traversal.property.suffix
                ]

                subject_uri = rules.prefixes.get(traversal.class_.prefix, rules.metadata.namespace)[
                    traversal.class_.suffix
                ]
                for value_type in cast(MultiValueTypeInfo, property_.value_type).types:

```